opencv/src/OpenCV/Core/ArrayOps.hs

```diff
@@ -1058,3 +1059,28 @@ perspectiveTransform srcPoints transformationMat = unsafePerformIO $
   where
     numPts   = fromIntegral $ V.length srcPoints
     c'numPts = fromIntegral $ V.length srcPoints
+
+-- | Find all non-zero points of a matrix.
+findNonZero
+    :: Mat shape ('S 1) depth
+    -> V.Vector Point2i
+findNonZero mat = unsafePerformIO $
+  withPtr mat $ \srcPtr ->
+  alloca $ \(numPointsPtr :: Ptr Int32) ->
+  alloca $ \(pointsPtrPtr :: Ptr (Ptr (Ptr C'Point2i))) -> do
```

Comment:
We need a `mask_` here to guarantee the finalization of the points array.

Reply:
Good catch. Just `mask_` the whole thing?

Reply:
Yes, a `mask_` before the `do` should do the right thing.